### PR TITLE
feat(sony projector): add pj talk protocol

### PR DESCRIPTION
**Pull request by stakach**
I'll write more specs, just so consider this WIP for now

### CODE REVIEW

File: drivers/sony/projector/pj_talk.cr [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    HDMI    = 0x0003 # same as InputB
+    InputA  = 0x0002
+    InputB  = 0x0003
+    InputC  = 0x0004
+    InputD  = 0x0005
+    USB     = 0x0006 # USB type B
+    Network = 0x0007 # network
+
+    def to_bytes : Bytes
+      Bytes[self.value >> 8, self.value & 0xFF]
```
Comment (kimburgess) [inline drivers/sony/projector/pj_talk.cr:76]:
```suggestion
      b = Bytes.new 2
      IO::ByteFormat::BigEndian.encode(value, b)
      b
```

File: drivers/sony/projector/pj_talk.cr [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    end
+
+    def self.from_bytes(b : Bytes)
+      Input.from_value((b[0].to_u16 << 8) + b[1])
```
Comment (kimburgess) [inline drivers/sony/projector/pj_talk.cr:80]:
```suggestion
      from_value IO::ByteFormat::BigEndian.decode(UInt16, b)
```

File: drivers/sony/projector/pj_talk.cr [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+  @community : String = ""
+
+  def on_load
+    # abstract tokenizer
+    transport.tokenizer = Tokenizer.new do |io|
+      bytes = io.to_slice
+
+      # Min message length is 8 bytes
+      # return the message length
+      bytes.size < 10 ? -1 : 10 + bytes[9]
```
[the remark this answers is not in the document]
Comment (stakach) [reply drivers/sony/projector/pj_talk.cr:29]: yeah, left over ruby comment where the tokeniser dropped the token match from the data, so there was a smaller message returned

File: drivers/sony/projector/pj_talk.cr [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    power?(priority: 50)
+  end
+
+  def power?(priority : Int32 = 0, **options)
+    do_send(:get, :power_status, **options, priority: priority).get
+    !!self[:power].try(&.as_bool)
+  end
+
+  enum Input
+    HDMI    = 0x0003 # same as InputB
```
[the remark this answers is not in the document]
Comment (stakach) [reply drivers/sony/projector/pj_talk.cr:67]: copied from the other sony driver, different protocol but the IDs for inputs were the same

**Pull request merged**